### ISSUE TITLE
Confusing status for successful pod

### ISSUE THREAD

**Issue by erictune** [kubernetes/kubernetes#14307]
```console
$ cat > succeeder.yaml <<EOF
apiVersion: v1
kind: Pod
metadata:
  name: succeeder
  labels:
spec:
  containers:
  - command:
    - bash
    - -c
    - echo ok
    image: centos:centos7
    imagePullPolicy: IfNotPresent
    name: c
  restartPolicy: OnFailure
EOF
$ kubectl create -f succeeder.yaml
pods/succeeder
$ sleep 10
$ kubectl describe pods/succeeder.yaml
...
Status:				Succeeded
...
Containers:
  c:
    State:		Terminated
      Reason:		Error
      Exit Code:	0
      Started:		Mon, 21 Sep 2015 13:55:06 -0700
      Finished:		Mon, 21 Sep 2015 13:55:06 -0700
```

It is confusing to call the container "Terminated with Reason Error" when Exit Code was 0, and when the Pod status is "Succeeded".

It is also confusing that there are so many events for the Pod, but not one saying that a successful Container exited 0 and was cleaned up.

**Comment by erictune** [comment]
Would be nice to sneak a fix into v1.1 since it makes Job easier to use, but may be too late for that.

**Comment by feihujiang** [comment]
Yes, and when use 
```
$ kubectl get pods succeeder

NAME      READY     STATUS    RESTARTS   AGE
succeeder  0/1       Error     0          22s
```
The STATUS result is Error, it is confusing too.

**Comment by bgrant0607** [comment]
My guess is that this was caused by #13727. We should fix this.
cc @jiangyaoguo @kubernetes/goog-node

**Comment by dchen1107** [comment]
Ok, I will take a quick look at this one.

**Comment by ncdc** [comment]
With #15082 in, is this fixed? See this comment specifically: https://github.com/kubernetes/kubernetes/pull/15082#issuecomment-145741026

**Comment by yujuhong** [comment]
Yes, this is fixed by #15082
